### ISSUE TITLE
Is it possible to make nonzero jittable

### ISSUE THREAD

**Issue by Joshuaalbert** [google/jax#4585]
Currently, it's not possible to do things like
```python
from jax import jit
import jax.numpy as jnp

jit(jnp.where(a>1))(jnp.arange(3))
```
because `nonzero` is not jittable. Is there some trick that would accomplish the same thing as `jnp.where(condition)`?

**Comment by Joshuaalbert** [comment]
Suppose also that you know how large the output should be, e.g. `jnp.where(a=jnp.max(a))` where `a` has a unique maximum.

**Comment by Joshuaalbert** [comment]
I guess something like,
```python
#jnp.where(a=jnp.max(a))
jnp.unravel_index(jnp.argmax(a.flatten()), a.shape)
```
So one way to use this for a general `jnp.where(condition)` is to use `unravel_index` on all elements and then to use a mask in any downstrea operations.

**Comment by mattjj** [comment]
Good question! There isn't a good trick to get the same thing as "unary-where" `jnp.where(condition)`, though I expect we'll add support for it to `jax.mask` soon.

Usually there is a good way to re-express the overall computation involving the unary-where to avoid its use, but it's a case-by-case thing.

For that reason I'm not sure if we'll be able to provide a more general answer to this question (though we're happy to help look at specific examples!). WDYT?

**Comment by Joshuaalbert** [comment]
This seem reasonable. My need was to select largest location in an array with one unique maximum, which I was able to do without unary-where as able. I don't know about `jax.mask`. What is its intended purpose?

**Comment by mattjj** [comment]
Indeed, we haven't documented `jax.mask`, because it's just a prototype! It's a tool for supporting certain kinds of shape polymorphism. We'll post a design doc or tutorial at some point in the next few months, so keep an eye out!

In the meantime I'll close this issue, but please don't hesitate to open others!
